panther_lights/src/driver_node.cpp

```diff
@@ -53,67 +52,114 @@ DriverNode::DriverNode(const std::string & node_name, const rclcpp::NodeOptions
 
+  chanel_2_ts_ = this->get_clock()->now();
   chanel_2_sub_ = std::make_shared<image_transport::Subscriber>(
     it->subscribe("lights/driver/channel_2_frame", 5, [&](const ImageMsg::ConstSharedPtr & msg) {
       FrameCB(msg, chanel_2_, chanel_2_ts_, "channel_2");
       chanel_2_ts_ = msg->header.stamp;
     }));
 
-  set_brightness_server_ = this->create_service<SetLEDBrightnessSrv>(
-    "lights/driver/set/brightness", std::bind(&DriverNode::SetBrightnessCB, this, _1, _2));
-
-  RCLCPP_INFO(this->get_logger(), "Initialized successfully.");
+  RCLCPP_INFO(this->get_logger(), "Initialized ImageTransport successfully.");
 }
 
 void DriverNode::OnShutdown()
 {
-  // Clear LEDs
+  ClearLEDs();
+
+  if (led_control_granted_) {
+    ToggleLEDControl(false);
+  }
+}
+
+void DriverNode::ClearLEDs()
+{
   chanel_1_.SetPanel(std::vector<std::uint8_t>(num_led_ * 4, 0));
   chanel_2_.SetPanel(std::vector<std::uint8_t>(num_led_ * 4, 0));
+}
 
-  // Give back control over LEDs
-  if (panels_initialised_) {
-    SetPowerPin(false);
+void DriverNode::ToggleLEDControl(const bool enable)
+{
+  auto request = std::make_shared<SetBoolSrv::Request>();
+  request->data = enable;
+
+  if (!enable_led_control_client_->wait_for_service(std::chrono::seconds(3))) {
+    throw std::runtime_error(
+      "Timeout occurred while waiting for service '" +
+      std::string(enable_led_control_client_->get_service_name()) + "'!");
   }
 
-  gpio_driver_.reset();
+  enable_led_control_client_->async_send_request(
+    request, std::bind(&DriverNode::ToggleLEDControlCB, this, std::placeholders::_1));
+
+  RCLCPP_DEBUG(
+    this->get_logger(), "Sent request toggling LED control to '%s'.", enable ? "true" : "false");
+}
+
+void DriverNode::ToggleLEDControlCB(rclcpp::Client<SetBoolSrv>::SharedFutureWithRequest future)
+{
+  RCLCPP_DEBUG(this->get_logger(), "Received response after toggling LED control.");
+
+  auto result = future.get();
+
+  auto request = result.first;
+  auto response = result.second;
+
+  if (request->data == true && response->success) {
+    led_control_granted_ = true;
+    ClearLEDs();
+    RCLCPP_INFO_THROTTLE(this->get_logger(), *this->get_clock(), 5000, "LED control granted.");
+  } else if (request->data == false && response->success) {
+    led_control_granted_ = false;
+    RCLCPP_INFO_THROTTLE(this->get_logger(), *this->get_clock(), 5000, "LED control revoked.");
+  } else {
+    RCLCPP_ERROR_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000, "Failed to toggle LED control.");
+  }
 }
 
 void DriverNode::FrameCB(
   const ImageMsg::ConstSharedPtr & msg, const apa102::APA102 & panel,
   const rclcpp::Time & last_time, const std::string & panel_name)
 {
+  if (!led_control_granted_) {
+    ToggleLEDControl(true);
+
+    auto message = "LED control not granted, ignoring frame!";
+    RCLCPP_WARN_STREAM_SKIPFIRST_THROTTLE(
+      this->get_logger(), *this->get_clock(), 5000, message << " on " << panel_name << "!");
+    return;
+  }
```

Comment:
Yep, initially we agreed on this solution but later came the issue with redesigning the initialization process so I kept the current form to prevent reimplementing this twice. 

Additionally, this is not going to be too quick since we should ensure the LED control is granted and what's more, periodically ensure that this state hasn't changed.